### ISSUE TITLE
RFC: Move linuxkit tool to linuxkit/tool?

### ISSUE THREAD

**Issue by rn** [linuxkit/linuxkit#2411]
The `linuxkit` tool is currently part of the `linuxkit/linuxkit` repository but is actually somewhat independent from the rest of the repository which contains the LinuxKit packaging and tooling. the `linuxkit` tool itself is more generic and could be used to run non-LinuxKit images as well. By moving it out of the repository we could also restructure it more easily to so that the core could be re-usable as Go package.

This issue is here to discuss the pros/cons of moving it.

**Comment by justincormack** [comment]
I don't think there are many downsides, we are already trying to move to getting people to install from eg homebrew, and in future (soon, hopefully) other package managers, and the Makefile currently already has to get `moby` from elsewhere so there is little change there.

**Comment by dave-tucker** [comment]
I think this would be a good idea. Could we perhaps also rename the tool at the same time?

There is confusion as we are overloading the name `linuxkit`.
It's both the distro and packages AND the tool used to run the output from `moby build`

**Comment by justincormack** [comment]
I don't know about renaming. That means more names to think of...

**Comment by dave-tucker** [comment]
We could always call it `lkt`

**Comment by deitch** [comment]
On holiday, so not much time to write but agree. Great idea.

**Comment by deitch** [comment]
And the build tool `mby`? :-)

**Comment by justincormack** [comment]
I don't think we have to do the rename at the same time, although calling the repo `tool` is a bit annoying when you clone it. I could maybe get used to `lkt`.

**Comment by ijc** [comment]
> calling the repo `tool` is a bit annoying when you clone it

FWIW I recently discovered that you can rename your clones. For example https://github.com/ijc/moby-tool is a clone I made of https://github.com/moby/tool which eas originally https://github.com/ijc/tool, but I renamed it in the `Settings` tab. There seem to have been no ill effects.

So the annoyance is limited to a one time step after you fork (and perhaps needing to remember which repos you do this to).

`lkt` is ok, mae me think of a throat sweet (whether that is good, bad or benign I don't know). Quite why I went there rather than to a necklace who knows...

**Comment by justincormack** [comment]
Ah I was just wondering if you could do that (I think I may have done it in the past). Maybe we could start with `tool` and not renaming and do that as another stage?

**Comment by ijc** [comment]
I don't follow the "not renaming" bit of what you say, what I was saying is that we can rename our personal forks to whatever we like irrespective of the name of the master/parent repo, so we have `linuxkit/tool` and my clone is `ijc/linuxkit-tool` and yours is `justcormack/bob` or whatever...

WRT the move itself I find it very convenient to be able to just type `make` and have the binaries appear in `$GOPATH/src/github.com/linuxkit/linuxkit/bin` (which is in my `$PATH` for this workspace). Are we planning to retain that i.e. by having `linuxkit/tool` be handled the same as `moby.tool` today (so clone and build)? I could also live with a target which did `go get -u ...` (as a convenience, I know I could just type it myself) since `$GOPATH/bin` is also in my `$PATH`, although having a known good hash for the given linuxkit tree is quite nice IMHO, at least while things aren't 1.0.

**Comment by justincormack** [comment]
By not renaming I meant not yet rename the tool from linuxkit to eg lkt

**Comment by justincormack** [comment]
Is someone volunteering to extract the history from linuxkit/linuxkit? Thats the most tedious part...

**Comment by rn** [comment]
I'm not...not sure I'm know where I would need to start...my git foo is not very strong

**Comment by justincormack** [comment]
I did it before with the `moby` split, can probably dig out the details.
